shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -175,14 +173,11 @@ - (void)setViewController:(FlutterViewController*)viewController {
   _viewController = [viewController getWeakPtr];
   self.iosPlatformView->SetOwnerViewController(_viewController);
   [self maybeSetupPlatformViewChannels];
+  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(onFlutterVCWillDeallocNotif:) name:FlutterViewControllerWillDealloc object:viewController];
```

Comment:
This isn't removing the observer for the previous viewController.
```objective-c
- (void)setViewController:(FlutterViewController*)viewController {
  if (_viewController == viewController) {
    return;
  }
  FML_DCHECK(self.iosPlatformView);
  [[NSNotificationCenter defaultCenter] removeObserver:self name:FlutterViewControllerWillDealloc object:_viewController];
   _viewController = [viewController getWeakPtr];
...
```

Reply:
> need to set CLANG_ENABLE_OBJC_WEAK flag

Nope, because we are using `__block` not `__weak`.